tests/regression/test_mesh_vertexonly.py

```diff
@@ -0,0 +1,333 @@
+from firedrake import *
+import pytest
+import numpy as np
+from mpi4py import MPI
+
+# Utility Functions
+
+
+def cell_midpoints(m):
+    """Get the coordinates of the midpoints of every cell in mesh `m`.
+
+    :param m: The mesh to generate cell midpoints for.
+
+    :returns: A tuple of numpy arrays `(midpoints, local_midpoints)` where
+    `midpoints` are the midpoints for the entire mesh even if the mesh is
+    distributed and `local_midpoints` are the midpoints of only the
+    rank-local non-ghost cells."""
+    if isinstance(m.topology, mesh.ExtrudedMeshTopology):
+        raise NotImplementedError("Extruded meshes are not supported")
+    m.init()
+    V = VectorFunctionSpace(m, "DG", 0)
+    f = Function(V).interpolate(m.coordinates)
+    # since mesh may be distributed, the number of cells on the MPI rank
+    # may not be the same on all ranks (note we exclude ghost cells
+    # hence using num_cells_local = m.cell_set.size). Below local means
+    # MPI rank local.
+    num_cells_local = m.cell_set.size
+    num_cells = MPI.COMM_WORLD.allreduce(num_cells_local, op=MPI.SUM)
+    local_midpoints = f.dat.data_ro
+    local_midpoints_size = np.array(local_midpoints.size)
+    local_midpoints_sizes = np.empty(MPI.COMM_WORLD.size, dtype=int)
+    MPI.COMM_WORLD.Allgatherv(local_midpoints_size, local_midpoints_sizes)
+    midpoints = np.empty((num_cells, m.cell_dimension()), dtype=float)
+    MPI.COMM_WORLD.Allgatherv(local_midpoints, (midpoints, local_midpoints_sizes))
+    assert len(np.unique(midpoints, axis=0)) == len(midpoints)
+    return midpoints, local_midpoints
+
+
+"""Parent meshes used in tests"""
+parentmeshes = [
+    pytest.param(UnitIntervalMesh(1), marks=pytest.mark.xfail(reason="swarm not implemented in 1d")),
+    UnitSquareMesh(1, 1),
+    pytest.param(ExtrudedMesh(UnitSquareMesh(1, 1), 1), marks=pytest.mark.xfail(reason="extruded meshes not supported")),
+    UnitCubeMesh(1, 1, 1)
+]
+
+
+"""Number of random coordinates"""
+ncoords = [0, 1, 100]
+
+
+def random_coords(n, gdim):
+    """
+    Get an array of `n` random coordinates with coordinate elements
+    between -0.5 and 1.5. The number of elements per coordinate is given
+    by `gdim`.
+    """
+    a, b = -0.5, 1.5
+    return (b - a) * np.random.random_sample(size=(n, gdim)) + a
```

Comment:
Deliver these as a fixture too:

```
@pytest.fixture(params=[1, 10, 100], ids=lambda x: f"{x}-coords")
def vertex_coords(request, parent):
    size = (request.param, parent.geometric_dimension())
    ...
```

Probably also use a consistent seed, otherwise these tests are not deterministic.

Reply:
Changed in 2bad0425e22a748a22313d7c84e5f8671a125e1b